project_euler/problem_102/sol102.py

```diff
@@ -0,0 +1,94 @@
+"""Project Euler Problem 102: https://projecteuler.net/problem=80
+
+"It is well known that if the square root of a natural number is not an
+integer, then it is irrational. The decimal expansion of such square
+roots is infinite without any repeating pattern at all.
+
+"The square root of two is 1.41421356237309504880..., and the digital
+sum of the first one hundred decimal digits is 475.
+
+"For the first one hundred natural numbers, find the total of the
+digital sums of the first one hundred decimal digits for all the
+irrational square roots."
+
+To solve this problem, create vectors from (0,0) to each of the points
+of the triangle. Then, add the angles between all the vectors. If the
+triangle includes the origin, the angles will add to 2*pi
+radians. Otherwise, they will add to less.
+
+References:
+ - https://stackoverflow.com/questions/2049582/
+
+"""
+import math
+import os
+
+
+class vector:
+    """
+    A simple vector class
+    """
+
+    def __init__(self, x: float, y: float) -> None:
+        """
+        Initialize a vector with x and y coordinates.
+
+        >>> isinstance(vector(1,2), vector)
+        True
+        """
+
+        self.x = x
+        self.y = y
+
+    def len(self) -> float:
+        """
+        Compute the length of the vector
+
+        >>> vector(3,4).len()
+        5.0
+        """
+        return math.sqrt(self.x * self.x + self.y * self.y)
+
+
+def dot(a: vector, b: vector) -> float:
+    """
+    Compute the dot product of two vectors
+
+    >>> dot(vector(1,2), vector(3,4))
+    11.0
+    """
+    return float(a.x * b.x + a.y * b.y)
+
+
+def solution() -> int:
+    """
+    Returns the number of triangles from the input that contains the origin.
+
+
+    >>> solution()
+    228
+    """
+
+    infile = open(os.path.join(os.path.dirname(__file__), "p102_triangles.txt"), "r")
+    inside = 0
+
+    for line in infile:
+        coords = line.strip().split(",")
+        vecs = []
+        for cid in range(3):
```

Comment:
Descriptive variable names than `coords`, `vecs`, `cid`, `a`, `b`?
What's even `cid`?

Reply:
Done. Apologies, in my field it's common to use "cid" (and, relatedly, "cidx") to iterate over arrays like "coords," giving "coord ID" or "coord index." I have fixed this.